src/hotspot/share/gc/g1/g1EvacuationFailureRegions.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2021, Huawei Technologies Co. Ltd. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_GC_G1_G1EVACUATIONFAILUREREGIONS_HPP
+#define SHARE_GC_G1_G1EVACUATIONFAILUREREGIONS_HPP
+
+#include "utilities/bitMap.hpp"
+#include "utilities/bitMap.inline.hpp"
+
+class HeapRegionClosure;
+class HeapRegionClaimer;
+
+// This class
+//     1. records for every region on the heap whether evacuation failed for it.
```

Comment:
Please write about the class without using a list; at most use an unordered list as there seems to be no ordering here (i.e. at most use bullets). I strongly prefer just two regular sentences.